engine/runtime-compiler/src/test/java/org/enso/compiler/test/mock/WithMockCompilerContext.java

```diff
@@ -0,0 +1,160 @@
+package org.enso.compiler.test.mock;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.List;
+import java.util.Set;
+import org.apache.commons.vfs2.FileObject;
+import org.enso.compiler.Compiler;
+import org.enso.compiler.context.CompilerContext.Module;
+import org.enso.compiler.data.CompilerConfig;
+import org.enso.editions.LibraryName;
+import org.enso.pkg.Package;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import scala.jdk.javaapi.CollectionConverters;
+
+/**
+ * A {@link TestRule} to use with the {@link MockCompilerContext}.
+ *
+ * <p>I recommend to use it as a {@link org.junit.Rule}, instead of {@link org.junit.ClassRule} to
+ * ensure that the whole context is reset before each test.
+ */
+public final class WithMockCompilerContext implements TestRule {
+  private final MockPackageRepository repo;
+  private final ByteArrayOutputStream out;
+  private final CompilerConfig compilerCfg;
+  private final MockCompilerContext compilerContext;
+  private final Compiler compiler;
+
+  private WithMockCompilerContext(ByteArrayOutputStream out, CompilerConfig compilerCfg) {
+    this.repo = MockPackageRepository.create();
+    this.out = out;
+    this.compilerCfg = compilerCfg;
+    this.compilerContext = new MockCompilerContext(repo, new PrintStream(out));
+    this.compiler = new Compiler(compilerContext, repo, compilerCfg);
+  }
+
+  public static WithMockCompilerContext createDefault() {
+    return new Builder().build();
+  }
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  public MockPackageRepository getRepo() {
+    return repo;
+  }
+
+  public Compiler getCompiler() {
+    return compiler;
+  }
+
+  public Package<FileObject> createPackage(LibraryName pkgName, SourceModule module) {
+    return repo.createPackage(pkgName, module);
+  }
+
+  /**
+   * Create a package with the given set of source modules in the virtual file system.
+   *
+   * @param pkgName
+   * @param modules
+   * @return Created package, or null if an error occurred.
+   */
+  public Package<FileObject> createPackage(LibraryName pkgName, Set<SourceModule> modules) {
```

Comment:
Why not have just a `Module` factory method?
```suggestion
  public Module createModule(QualifiedName name, String context) {
```
which would put the module to the right `Package` based on the provided FQN `name`?

Reply:
Refactored in 57e479ee5cb96251ac9d41faeed49d07c5d59843